public/app/features/plugins/extensions/registry/Registry.ts

```diff
@@ -13,7 +13,7 @@ export type PluginExtensionConfigs<T> = {
 export type RegistryType<T> = Record<string | symbol, T>;
 
 // This is the base-class used by the separate specific registries.
-export abstract class Registry<TRegistryValue, TMapType> {
+export abstract class Registry<TRegistryValue extends object | unknown[] | Record<string | symbol, unknown>, TMapType> {
```

Comment:
Maybe we could extract this to it's own type as we do on line 13?

Reply:
That's sad 😢  let's keep as is then

Reply:
Yeah it would be nice to get it resolved in this scenario. Or is that a big problem, wdyt?